docs/features/software-templates/writing-templates.md

```diff
@@ -477,6 +477,34 @@ template can be published to multiple providers.
 Note, that you will need to configure an [authentication provider](../../auth/index.md#configuring-authentication-providers), alongside the
 [`ScmAuthApi`](../../auth/index.md#scaffolder-configuration-software-templates) for your source code management (SCM) service to make this feature work.
 
+### The Repository Branch Picker
+
+Similar to the repository picker, there is a picker for branches to support autocompletion. A full example could look like this:
+
+```yaml
+- title: Choose a branch
+  required:
+    - repoBranch
+  properties:
+    repoBranch:
+      title: Repository Branch
+      type: string
+      ui:field: RepoBranchPicker
+      ui:options:
+        requestUserCredentials:
+          secretsKey: USER_OAUTH_TOKEN
+```
+
+Passing the `requestUserCredentials` object is required for autocompletion to work.
+If you're also using the repository picker, you should simply duplicate this part from there.
```

Comment:
Is there a way to share integration configuration between extensions @backstage/scaffolder-maintainers?

Reply:
This has been discussed on Discord as well: https://discord.com/channels/687207715902193673/923143744176418836/1252243951633567754

TL;DR: @benjdlambert has an open [RFC](https://github.com/backstage/backstage/issues/16996) that would allow this, but as of right now, this is not possible yet.